corehq/apps/reports/filters/users.py

```diff
@@ -315,13 +316,18 @@ def user_es_query(cls, domain, mobile_user_and_group_slugs):
     @classmethod
     def pull_users_and_groups(cls, domain, mobile_user_and_group_slugs,
                               include_inactive=False, limit_user_ids=None):
-        user_ids = cls.selected_user_ids(mobile_user_and_group_slugs)
+        user_ids = set(cls.selected_user_ids(mobile_user_and_group_slugs))
         user_types = cls.selected_user_types(mobile_user_and_group_slugs)
         group_ids = cls.selected_group_ids(mobile_user_and_group_slugs)
+        location_ids = cls.selected_location_ids(mobile_user_and_group_slugs)
         users = []
 
+        if location_ids:
+            for location_id in location_ids:
+                user_ids |= set(get_user_ids_by_location(domain, location_id))
```

Comment:
ah right, always get tripped up by that function name. maybe a convention that suggests that this isn't the method you want to use most of the time (my assumption is that most of the time you want to get all user_ids for that location and its children, but that could be wrong).